### PR TITLE
Add libudev 1.6.11 for use with Dolphin

### PR DESCRIPTION
**Pull request by Makman2**
Addresses https://github.com/flathub/org.libretro.RetroArch/issues/115

This is an extremely hacky workaround in my opinion, however it works. It adds the most recent libudev (v1.6.11) to the flatpak so the Dolphin core can properly load.

Why is it hacky? I am pretty new to flatpak, and trying thousands (though probably not all :P) options to modify the systemd build, I ended up putting in some patches to in-place-modify systemd's meson build file to set some paths (I haven't seen an option from flatpak to pass parameters to the meson command).
Also I am not sure if the installation like I did of systemd is okay like it is. I mean it's something rather linux-internal, not sure if that makes problems to install the stuff somewhere not default.

However, if someone is looking for a quick-fix, this one is going to work, however building the retroarch flatpak is required. But it wasn't too hard for me at least :)

In any case: If you have suggestions for improvement, let me know :) (for real, they are eagerly welcome :sweat_smile: )

### CODE REVIEW

File: udev/systemd-udev.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+    "name": "udev",
+    "sources": [
+        {
+            "type": "archive",
+            "url": "https://github.com/systemd/systemd/archive/v242.tar.gz",
```
Comment (TingPing) [inline udev/systemd-udev.json]: I'm confident there is a ton of `cleanup` missing here.

File: udev/set-paths.patch [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
++                'localstatedir=/app/var',
+         ],
+         meson_version : '>= 0.46',
+        )
+@@ -70,7 +70,7 @@ endif
+ conf.set10('HAVE_SPLIT_BIN', split_bin,
+            description : 'bin and sbin directories are separate')
+ 
+-rootprefixdir = get_option('rootprefix')
++rootprefixdir = '/app'
```
Comment (TingPing) [inline udev/set-paths.patch]: Why are you patching this instead of just `-Drootprefix=/app`?

File: udev/set-paths.patch [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+index 3593eac710..2582dcab8c 100644
+--- a/meson.build
++++ b/meson.build
+@@ -6,8 +6,8 @@ project('systemd', 'c',
+         default_options: [
+                 'c_std=gnu99',
+                 'prefix=/usr',
+-                'sysconfdir=/etc',
+-                'localstatedir=/var',
++                'sysconfdir=/app/etc',
```
Comment (TingPing) [inline udev/set-paths.patch]: Same as other comment.

**Comment by TingPing** [comment]
>  (I haven't seen an option from flatpak to pass parameters to the meson command).

`"config-opts": ["-Dfoo=bar"]`

File: udev/systemd-udev.json (hunk 1 above)
Comment (Makman2) [reply udev/systemd-udev.json]: Yeah very likely :P I'm looking into it what's cleanable today or the next days, schedule gets a little bit stuffed right now :+1:

File: udev/set-paths.patch (hunk 2 above)
Comment (Makman2) [reply udev/set-paths.patch]: I actually tried via `config-opts` to pass those, but that didn't seem to work out. But I'll test it again to be sure, not that I just missed something or did a typo or so...

Comment (Makman2) [reply udev/set-paths.patch]: Okay it works with `config-opts`... phew... And I know why it didn't work before: it's supposed to be an array, not a single string... flatpak just gives a warning at top, which I evidently missed...

**Comment by Makman2** [comment]
Are there any committing guidelines for this repo? Specifically, do you prefer pushing fixup commits on top of this PR or is an amend preferred? (I usually prefer amends for a cleaner history^^)

**Comment by TingPing** [comment]
Personally I'd say do small new commits. At merge time it can be squashed.

File: udev/systemd-udev.json [hunk 4]
```diff
@@ -1,14 +1,39 @@
...
+        "/share/doc",
+        "/share/factory",
+        "/share/locale",
+        "/share/man",
+        "/share/pkgconfig/systemd.pc",
+        "/share/polkit-1/actions",
+        "/share/systemd",
+        "/share/zsh",
+        "*.d"
+    ],
```
Comment (TingPing) [inline udev/systemd-udev.json]: This list is probably good, I would ask to look at `meson_options.txt` to see if some of them can be not built entirely though.

**Comment by Makman2** [comment]
The only promising option I found was `static-libsystemd`, but that hadn't a reall effect on the build bartifact (setting it to false). I guess that's the reason why eudev exists, because systemd's version is pretty tightly integrated into the build with no efforts to allow a libudev-only build^^

**Comment by Makman2** [comment]
In any case, the libsystemd build doesn't appear to take very long, so for now I think that's fine :+1: :)

**Comment by TingPing** [comment]
:+1:

File: org.libretro.RetroArch.json [hunk 5]
```diff
@@ -23,7 +23,7 @@
     "shared-modules/SDL/SDL_mixer-1.2.12.json",
     "shared-modules/SDL/SDL_net-1.2.8.json",
     "shared-modules/SDL/SDL_ttf-2.0.11.json",
-    "shared-modules/udev/udev-175.json",
+    "udev/systemd-udev.json",
```
Comment (RobLoach) [inline org.libretro.RetroArch.json]: Would it be worthwhile pushing this updated udev to [shared-modules/udev](https://github.com/flathub/shared-modules/tree/master/udev)?

Comment (TingPing) [reply org.libretro.RetroArch.json]: I'm not sure, I'm a bit annoyed that it requires building systemd compared to eudev. So I don't know which is a better option.

**Comment by Makman2** [comment]
Should the issue be closed at https://github.com/flathub/org.libretro.RetroArch/issues/115 or is additional testing needed (e.g. by other users)? :)

**Pull request merged**